handbook/technical-overview/x-road/SecurityServerInstallationAndRegistrationSteps.md

```diff
@@ -0,0 +1,315 @@
+
+Enter PIN Code  
+Then Press LOG IN
+
+![](../../../.gitbook/assets/x-road/12.png)
+
+The red error message bar disappears
+
+Next is Time stamping.   
+Go to: Settings &gt; Timestamping Services &gt; ADD
+
+![](../../../.gitbook/assets/x-road/13%20%281%29.png)
+
+Pick time-stamping service from the list.
+
+Press OK.
+
+![](../../../.gitbook/assets/x-road/14%20%281%29.png)
+
+Timestamping message added will appears.
+
+![](../../../.gitbook/assets/x-road/15.png)
+
+Next step is to add Keys and Certificates  
+Open KEYS AND CERTIFICATES
+
+![](../../../.gitbook/assets/x-road/16%20%281%29.png)
+
+Press ADD KEY  
+Start making sign key
+
+![](../../../.gitbook/assets/x-road/17.png)
+
+Enter ”sign”  
+Choose SIGNING And Client  
+Change to PEM in CSR Format
+
+![](../../../.gitbook/assets/x-road/18.png)
+
+Press GENERATE CSR
+
+![](../../../.gitbook/assets/x-road/19%20%281%29.png)
+
+Press DONE
+
+![](../../../.gitbook/assets/x-road/20%20%281%29.png)
+
+![](../../../.gitbook/assets/x-road/21%20%281%29.png)
+
+The certificate request is downloaded to browser's download folder.
+
+Now make auth key  
+Enter “auth” and press NEXT
+
+![](../../../.gitbook/assets/x-road/22.png)
+
+Choose AUTHENTICATON and change CSR Format to PEM
+
+![](../../../.gitbook/assets/x-road/23.png)
+
+Enter your Server DNS name \(CN\)
+
+![](../../../.gitbook/assets/x-road/24%20%281%29.png)
+
+Press GENERATE CSR
+
+![](../../../.gitbook/assets/x-road/25.png)
+
+The certificate request is downloaded to browser's download folder.
+
+![](../../../.gitbook/assets/x-road/26%20%281%29.png)
+
+Now you can see that there are two keys in the overview, Sign and Auth.
+
+VANTAR – Senda póst á Begga
+
+### Importing signed certifications
+
+
+Open KEYS AND CERTIFICATIONS and press IMPORT CERT.
+
+![](../../../.gitbook/assets/x-road/27.png)
+
+![](../../../.gitbook/assets/x-road/28%20%281%29.png)
+
+Activate auth signed certificate, needs to click the name of the certificate \(test.xrd.island.is...\) and press Activate
+
+
+SCREENSHOT NEEDED
+
+![](../../../.gitbook/assets/x-road/29.png)
+
+![](../../../.gitbook/assets/x-road/30.png)
+
+![](../../../.gitbook/assets/x-road/31%20%281%29.png)
+
+Finally press **Register** on the auth certificate and enter inn the FQDN of the server and press ADD
+
+![](../../../.gitbook/assets/x-road/32%20%281%29.png)
+
+![](../../../.gitbook/assets/x-road/33.png)
+
+![](../../../.gitbook/assets/x-road/34%20%281%29.png)
+
+### Confirm communication between two security servers
+
+```python
+curl --insecure -H "X-Road-Client: IS-TEST/COM/5302922079/Origo-client" "
+https://origo-staging.xroad.coldcloudlab.com/r1/IS-TEST/GOV/7005942039/VMST-Protected/APIS/company?name=origo
+"
+```
+
+IS-**DEV**
+
+**Ísland.is to Skatturinn**:
+
+```python
+curl -H "X-Road-Client: IS-DEV/GOV/10000/island-is-client" "http://localhost:8080/r1/IS-DEV/GOV/10006/Skatturinn-Protected/APIS-v1/company?name=skatturinn"
+```
+
+IS-**TEST**
+
+**Ísland.is to Skatturinn:**
+
+```python
```

Comment:
`bash` instead of python in this file

Reply:
Sounds good to keep the assets close to the documentation they relate to.